subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AnnotationProcessorDetector.java

```diff
@@ -113,7 +115,9 @@ public Boolean calculate(File file, FileType fileType) {
                         zipFile.close();
                     }
                 } catch (IOException e) {
-                    throw new UncheckedIOException("Could not read service definition from JAR " + file, e);
+                    // Using a warning, not a deprecation message, because we already show a deprecation message during
+                    // compile classpath snapshotting, but we still want to warn
+                    LOGGER.warn("Could not read service definition from JAR " + file);
```

Comment:
Correct... This is a bit annoying here since we don't know when we log the error if it's for annotation or compile classpath. And the same jar will always trigger 2 deprecation warnings if it's on the compile classpath. That's not really user friendly, especially because the user doesn't care we fail detecting an annotation processor if they don't use it. Do you have a suggestion how to improve this?